CMakePresets.json

```diff
@@ -0,0 +1,134 @@
+{
+  "version": 3,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 20,
+    "patch": 0
+  },
+  "configurePresets": [
+    {
+      "name": "community",
+      "displayName": "Build ArangoDB Community Edition",
+      "binaryDir": "${sourceDir}/build/${presetName}",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "USE_ENTERPRISE": "Off",
+        "USE_JEMALLOC": "On",
+        "USE_IPO": "On"
+      }
+    },
+    {
+      "name": "enterprise",
+      "inherits": "community",
+      "displayName": "Build ArangoDB Enterprise Edition",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "USE_ENTERPRISE": "On",
+        "USE_JEMALLOC": "On",
+        "USE_IPO": "On"
+      }
+    },
+    {
+      "name": "community-developer",
+      "inherits": "community",
+      "displayName": "Build ArangoDB Community Edition (Developer Build)",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+        "USE_IPO": "Off",
+        "USE_MAINTAINER_MODE": "On",
+        "USE_FAIL_ON_WARNINGS": "On",
+	"USE_GOOGLE_TESTS": "On",
+	"USE_FAILURE_TESTS": "On",
+	"USE_STRICT_OPENSSL_VERSION": "Off",
+        "USE_SEPARATE_REPLICATION2_TESTS_BINARY": "On"
```

Comment:
Why is it default?

Reply:
I use make -j 18 to build all targets, but I don't need separate binary for replication2.

I think better to make same for replication2 as for iresearch.
You always create target but it will be builded only with explicit call